### PR TITLE
Align timestampWrites with WebGPU spec

### PR DESCRIPTION
**Pull request by beaufortfrancois**
Following updates in https://bugs.chromium.org/p/dawn/issues/detail?id=1800 and https://github.com/webgpu-native/webgpu-headers/pull/228, this PR aligns timestampWrites with the WebGPU spec.

### CODE REVIEW

**Review by kainino0x — APPROVED** [review]
LGTM but needs a code size rebaseline (I'll do it)

`./test/runner 'other.test_minimal_runtime_code_size_*' --rebaseline`

**Comment by kainino0x** [comment]
I think the rebaseline was not actually needed, just had to merge changes from main.

File: src/library_webgpu.js [hunk 1]
```diff
@@ -1535,36 +1527,27 @@ var LibraryWebGPU = {
...
-      for (var i = 0; i < count; ++i) {
-        timestampWrites.push(makeComputePassTimestampWrite(twPtr + {{{ C_STRUCTS.WGPUComputePassTimestampWrite.__size__ }}} * i));
-      }
-      return timestampWrites;
-    }
-
     if (descriptor) {
       {{{ gpu.makeCheckDescriptor('descriptor') }}}
-      desc = {};
+      var desc = {
```
Comment (kainino0x) [inline src/library_webgpu.js:1543]: `WARNING - [JSC_REDECLARED_VARIABLE] Redeclared variable: desc`
```suggestion
      desc = {
```

**Comment by beaufortfrancois** [comment]
@kainino0x Can you publish a new release of emscripten with those changes?

**Comment by sbc100** [comment]
@beaufortfrancois its mostly arbitrary when we do an emscripten release, but they tend to be every few weeks.  Since we just did one a couple of days back I would normally wait a week or two before doing another one.

Do you know about `./emsdk install tot` to get a tip-of-tree versions?  (or do you need an official release for some reason?)

**Comment by beaufortfrancois** [comment]
Thanks for the reminder @sbc100! I've communicated the `./emsdk install tot` handy tool to people who may need it.

We can wait then.

**Pull request merged**